karton/dashboard/graph/graph.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import annotations
+
+from typing import Callable, Dict, List, Optional, Set, Union
+
+from karton.core.inspect import KartonState
+from networkx import DiGraph, generate_gexf
+from networkx.readwrite.json_graph import adjacency_data, adjacency_graph
+
+NODE_SIZE: Callable[
+    [DiGraph, str], float
+] = lambda graph, identity: 65 + 3.5 * graph.in_degree(identity)
+DEFAULT_OPTIONS = {"color": {"r": 51, "g": 153, "b": 243, "a": 0}, "size": NODE_SIZE}
+EMPTY_METADATA = {"version": "none", "info": "none"}
+OPTIONS = ["color", "size"]
+
+
+class KartonNode:
+    def __init__(
+        self, identity: str, metadata: dict[str, str], filters, outputs
+    ) -> None:
+        self.identity = identity
+        self.metadata = metadata
+        self.filters = filters
+        self.outputs = outputs
+
+    def filter_contained(self, filter: Dict[str, str], output: Dict[str, str]) -> bool:
+        return all(item in output.items() for item in filter.items())
+
+    def __contains__(self, other: KartonNode) -> bool:
+        if self.filters and other.outputs:
+            for filter in self.filters:
+                if any(
+                    self.filter_contained(filter, output) for output in other.outputs
+                ):
+                    return True
+        return False
+
+
+class KartonGraph:
+    def __init__(self, state: KartonState) -> None:
+        self.state: KartonState = state
+        self.nodes: List[KartonNode] = []
+        self.graph: Dict[str, Set[str]] = {}
+
+    def style_nodes(
+        self,
+        graph: DiGraph,
+        options: Optional[
+            Dict[str, Union[Dict[str, int], Callable[[DiGraph, str], float]]]
+        ] = None,
+    ) -> None:
+        """Style the graph.
+
+        :param graph: a graph object
+        :type graph: networkx.DiGraph
+        :param options: styling options, which consists of:
+        \t`color`: a dictionary specifying RGBA
+        \t`size`: a function that takes a DiGraph and a node identity as an input and returns a size (`float`) 
```

Comment:
```suggestion
        \t`size`: a function that takes a DiGraph and a node identity as an input and returns a size (`float`)
```
This space seems to be making the linter angry